src/packages/fetch-engine/src/download.ts

```diff
@@ -92,6 +92,14 @@ export async function download(options: DownloadOptions): Promise<BinaryPaths> {
         'Warning',
       )} Precompiled binaries are not available for ${platform}. Read more about building your own binaries at https://pris.ly/d/build-binaries`,
     )
+  } else if (
+    os.platform === 'darwin' &&
+    os.arch === 'arm64' &&
+    EngineTypes.libqueryEngineNapi in options.binaries
+  ) {
+    throw new Error(
+      `Node-API is currently not supported for M1. Please remove \`nApi\` from your previewFeatures or remove the ENV VAR \`PRISMA_FORCE_NAPI\` `,
```

Comment:
```suggestion
      `Node-API is currently not supported for Apple M1. Please remove \`nApi\` from your previewFeatures or remove the environment variable \`PRISMA_FORCE_NAPI\``,
```

Did we try this on m1?

Also just curious: PRISMA_FORCE_NAPI is not used by users as far as we know?

Reply:
Yep I tried this on an M1 it will throw during `generate` but I've also added a check to throw during runtime

Reply:
Should we remove the mention of the ENV VAR?

Reply:
Maybe? PRISMA_FORCE_NAPI will go away soon anyway too?

Reply:
True I'll remove it